packages/features/ee/organizations/pages/settings/admin/AdminOrgPage.tsx

```diff
@@ -4,17 +4,29 @@ import { Trans } from "next-i18next";
 import { useState } from "react";
 
 import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { TeamRepository } from "@calcom/lib/server/repository/team";
 import { trpc } from "@calcom/trpc/react";
 import { Badge, ConfirmationDialogContent, Dialog, DropdownActions, showToast, Table } from "@calcom/ui";
 
 import { subdomainSuffix } from "../../../../organizations/lib/orgDomains";
 
 const { Body, Cell, ColumnTitle, Header, Row } = Table;
 
-export function AdminOrgTable() {
+type PageProps = {
+  ssrProps?: {
+    allOrgs?: Awaited<ReturnType<typeof TeamRepository.getAllOrgs>>;
+  };
+};
+export function AdminOrgTable({ ssrProps }: PageProps) {
   const { t } = useLocale();
   const utils = trpc.useUtils();
-  const [data] = trpc.viewer.organizations.adminGetAll.useSuspenseQuery();
+
+  let data = ssrProps?.allOrgs;
+  if (!data) {
```

Comment:
Not a fan of conditionally calling hooks here. I wonder if its worth improving performance here while were still on pages. It'd be great to just ships this app dir page without the need to support both

Reply:
Shall we just enable or disable the query based on if ssrProps.allOrgs is present?

Reply:
> Shall we just enable or disable the query based on if ssrProps.allOrgs is present?

`useSuspenseQuery` doesn't support enabled prop though :(